### DISCUSSION TITLE
Storage tables missing?

### DISCUSSION THREAD

**Discussion by RuthNjeri** [microsoft/AzUrlShortener#437] [Q&A] [answered]
Hello, I deployed the project but the storage 'urldatav3y7mstg' does not contain any tables, the documentation says that `There are two tables that will be automatically created at the first call.` which call should I make? An API one? How is this done? Thanks in advance.

**Comment by fboucher** [comment, marked as answer]
Good point!
@RuthNjeri are you planning to use the [TinyBlazorAdmin](https://github.com/FBoucher/tinyblazoradmin) as frontend? If yes, the tables should be created during your first short URL creation.

But now that this project has only the Redirect we should add an init method to create the tables. This method would only work once. THat would be required only when TinyBlazorAdmin is not used.

**Reply by stephexcel2004** [reply]
I created the table and field manually. 

Call the function api/{shortUrl} where the shortUrl is the RowKey value created in the table.

However, it re-direct to the default url (azure.com), not the long url.

![image](https://github.com/microsoft/AzUrlShortener/assets/37499187/734f8975-d9e8-48f8-8ffe-e70acb2d32aa)

I don't understand the init method you meant. I am not using tinyblazoradmin. Please help.

**Reply by stephexcel2004** [reply]
I got the table created using postman
xxx/api/UrlCreate

body:
{
    "url": "http://www.frankysnotes.com/2020/03/reading-notes-416.html",
    "vanity": "rn-416"
}

the table will be created

**Comment by RuthNjeri** [comment]
Thanks @FBoucher  with the TinyBlazor it worked, I'm now creating the URLS and each time I click on the shortened version, I get an error, bad request.

![image](https://user-images.githubusercontent.com/7622875/205001208-6064d79c-2d40-4673-a0ec-35e882c3f960.png)

**Reply by fboucher** [reply]
It's normal @RuthNjeri. Until a custom domain is added and DNS Rules configured the API is only available at
`[appName].azurewebsites.net/api/UrlRedirect/{vanity}`

Have a look at this doc page in the wiki: [How to Add a Custom Domain)](https://github.com/FBoucher/AzUrlShortener/wiki/How-to-Add-a-Custom-Domain)
